### DISCUSSION TITLE
question about Dock system

### DISCUSSION THREAD

**Discussion by smbss1** [mikke89/RmlUi#572] [General]
Hello, I found RmlUI today and I really appreciate it.
I wonder if there is a docking system in this library ?
Because I want to use it for a game engine editor but I did not find any information about docking windows.

**Comment by mikke89** [comment]
Thanks! I hope you'll enjoy the library.

We don't have any builtin docking system, so you will have to build it yourself. It will certainly be some effort, but the CSS/HTML coupled with some event handling is very flexible, so it should certainly be doable from user code.

Keep us updated on the progress, I'd be interested to hear how it works out, or if you have any questions. If you do make something that is reusable, perhaps we could consider integrating it into the library.

**Reply by ProgrammerSmarty** [reply]
I also want to create an engine with a dockable interface.

Building this into the library would be a great feature. I could always contribute and help with this if you want :). It would take a lot of work, but if loads of people want it and if it will make RmlUi even better, then... maybe this could be something someday.

**Reply by mikke89** [reply]
I'd be very interested to see what you can come up with! Contributions are always welcome. We'll have to discuss whether it fits for inclusion directly in the library, but in any case an implementation or prototype would be very valuable to use as a reference for others too. Let me know if I can help out in any way.

**Reply by ProgrammerSmarty** [reply]
I think it would fit in the library. I mean, all it needs to be is a property which lets you set the dock space relative to the window, like any other library that implements this. Maybe a "docking-enabled" property for the body and "dock" property for a container element might be viable? We might need a whole new element to have ease-of-use docking.

I mean, I don't really know. This could be a great feature if it goes through, but it's all up to you. I personally think this would do great things for this library and make UI layout creation way easier than it is now.

**Reply by ProgrammerSmarty** [reply]
I would be happy to pull request :)

**Reply by mikke89** [reply]
Yes, I am very much open to the idea. I just mean that I would like to see a prototype before saying for certain that it fits as part of the core library. I'd be very happy to see a pull request!
